packages/desktop-client/src/components/table.tsx

```diff
@@ -796,15 +797,33 @@ export function TableHeader({
   );
 }
 
-export function SelectedItemsButton({ name, items, onSelect }) {
-  const selectedItems = useSelectedItems();
+type SelectedItemsButtonNames = 'transactions' | 'schedules';
+
+type SelectedItemsButtonProps<T extends MenuItem = MenuItem> = {
+  name: SelectedItemsButtonNames;
+  items: Array<T | typeof Menu.line>;
+  onSelect: (name: string, items: Array<T | typeof Menu.line>) => void;
+};
+
+export function SelectedItemsButton<T extends MenuItem = MenuItem>({
+  name,
+  items,
+  onSelect,
+}: SelectedItemsButtonProps<T>) {
+  const { t } = useTranslation();
+  const selectedItems = useSelectedItems() as Set<T>;
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** manual type coercions lead to subtle and hard-to-spot bugs. Instead the type should be passed as an arg and then patched upstream.
```suggestion
  const selectedItems = useSelectedItems<T>();
```

Reply:
I don't think my TypeScript knowledge goes far enough to fix this all the way upstream.

This would allow your suggestion to pass type checks:
```
export function useSelectedItems<T = unknown>() {
  return useContext<Set<T>>(SelectedItems);
}
```
However it fails with `useContext`:
```
Argument of type 'Context<Set<unknown>>' is not assignable to parameter of type 'Context<Set<T>>'.
  The types of 'Provider.propTypes.value' are incompatible between these types.
    Type 'Validator<Set<unknown>>' is not assignable to type 'Validator<Set<T>>'.
      Type 'Set<unknown>' is not assignable to type 'Set<T>'.ts(2345)
```

Maybe someone else has an idea for this?

Reply:
Patched it for you. Hope you don't mind.

Reply:
Here's the patch: https://github.com/actualbudget/actual/compare/matiss/i18n-patch?expand=1

Feel free to apply it to this branch

Reply:
Hmm the patch helped but it still fails:
https://github.com/actualbudget/actual/actions/runs/10178799474/job/28153164815?pr=3036